jlm/hls/util/view.cpp

```diff
@@ -360,64 +346,110 @@ region_to_dot(rvsdg::Region * region)
+        && rvsdg::TryGetOwnerNode<rvsdg::ThetaNode>(*region->result(i)->output()))
     {
+      auto theta = rvsdg::TryGetOwnerNode<rvsdg::ThetaNode>(*region->result(i)->output());
       auto loopvar = theta->MapOutputLoopVar(*region->result(i)->output());
       dot << edge(loopvar.pre, loopvar.post, true);
     }
   }
 
-  if (region->nresults() > 1)
-  {
-    // order results horizontally
-    dot << "{rank=sink; ";
-    for (size_t i = 0; i < region->nresults(); ++i)
-    {
-      if (i > 0)
-      {
-        dot << " -> ";
-      }
-      dot << get_dot_name(region->result(i));
-    }
-    dot << "[style = invis]}\n";
-  }
-
   dot << "}\n";
 
   return dot.str();
 }
 
 std::string
-to_dot(rvsdg::Region * region)
+to_dot(
+    rvsdg::Region * region,
+    std::unordered_map<rvsdg::output *, std::string> & o_color,
+    std::unordered_map<rvsdg::input *, std::string> & i_color)
 {
   std::ostringstream dot;
   dot << "digraph G {\n";
-  dot << jlm::hls::region_to_dot(region);
+  dot << region_to_dot(region, o_color, i_color);
   dot << "}\n";
   return dot.str();
 }
 
 void
-view_dot(rvsdg::Region * region, FILE * out)
+view_dot(
+    rvsdg::Region * region,
+    FILE * out,
+    std::unordered_map<rvsdg::output *, std::string> & o_color,
+    std::unordered_map<rvsdg::input *, std::string> & i_color)
 {
-  fputs(jlm::hls::to_dot(region).c_str(), out);
+  fputs(to_dot(region, o_color, i_color).c_str(), out);
   fflush(out);
 }
 
 void
-dump_dot(jlm::llvm::RvsdgModule & rvsdgModule, const std::string & file_name)
+view_dot(rvsdg::Region * region, FILE * out)
+{
+  std::unordered_map<rvsdg::output *, std::string> o_color;
+  std::unordered_map<rvsdg::input *, std::string> i_color;
+  view_dot(region, out, o_color, i_color);
+}
+
+void
+dump_dot(llvm::RvsdgModule & rvsdgModule, const std::string & file_name)
+{
+  dump_dot(&rvsdgModule.Rvsdg().GetRootRegion(), file_name);
+}
+
+void
+dump_dot(
+    llvm::RvsdgModule & rvsdgModule,
+    const std::string & file_name,
+    std::unordered_map<rvsdg::output *, std::string> & o_color,
+    std::unordered_map<rvsdg::input *, std::string> & i_color)
+{
+  dump_dot(&rvsdgModule.Rvsdg().GetRootRegion(), file_name, o_color, i_color);
+}
+
+void
+dump_dot(rvsdg::Region * region, const std::string & file_name)
+{
+  auto dot_file = fopen(file_name.c_str(), "w");
+  view_dot(region, dot_file);
+  fclose(dot_file);
+}
+
+void
+dump_dot(
+    rvsdg::Region * region,
+    const std::string & file_name,
+    std::unordered_map<rvsdg::output *, std::string> & o_color,
+    std::unordered_map<rvsdg::input *, std::string> & i_color)
 {
   auto dot_file = fopen(file_name.c_str(), "w");
-  jlm::hls::view_dot(&rvsdgModule.Rvsdg().GetRootRegion(), dot_file);
+  view_dot(region, dot_file, o_color, i_color);
   fclose(dot_file);
 }
 
+void
+dot_to_svg(const std::string & file_name)
+{
+  auto cmd = "dot -Tsvg -O " + file_name;
+  if (system(cmd.c_str()))
+    exit(EXIT_FAILURE);
+}
+
```

Comment:
This can be done in bash, no? Why do we need this here in code, when it only does and external system call?